app/scripts/components/analysis/results/chart-card.tsx

```diff
@@ -17,20 +21,60 @@ import {
   CardActions,
   CardBody
 } from '$components/common/card';
-import { TimeseriesData } from './timeseries-data';
 import { ChartLoading } from '$components/common/loading-skeleton';
-import { ChartCardAlert, ChartCardNoData } from './chart-card-message';
-import Chart from '$components/common/chart/analysis';
+import Chart, { AnalysisChartRef } from '$components/common/chart/analysis';
 import { dateFormatter } from '$components/common/chart/utils';
+import { Tip } from '$components/common/tip';
+import { composeVisuallyDisabled } from '$utils/utils';
 
 interface ChartCardProps {
   title: React.ReactNode;
   chartData: TimeseriesData;
 }
 
+const ChartDownloadButton = composeVisuallyDisabled(ToolbarIconButton);
+
+const getNoDownloadReason = ({ status, data }: TimeseriesData) => {
+  if (status === 'errored') {
+    return 'Data loading errored. Download is not available.';
+  }
+  if (status === 'loading') {
+    return 'Download will be available once the data finishes loading.';
+  }
+  if (status === 'succeeded' && !data.timeseries.length) {
+    return 'There is no data to download.';
+  }
+  return '';
+};
+
 export default function ChartCard(props: ChartCardProps) {
   const { title, chartData } = props;
-  const { status, meta, data, error } = chartData;
+  const { status, meta, data, error, id } = chartData;
+
+  const chartRef = useRef<AnalysisChartRef>(null);
+
+  const noDownloadReason = getNoDownloadReason(chartData);
+
+  const onExportClick = useCallback(() => {
+    if (
+      !chartRef.current?.instanceRef.current ||
+      !chartData.data?.timeseries.length
+    )
+      return;
+
+    // Get start and end dates from chart instance.
+    const { dataStartIndex, dataEndIndex } = chartRef.current.instanceRef
+      .current.state as any;
+    // The indexes expect the data to be ascending, so we have to reverse the
+    // data.
+    const data = reverse(chartData.data.timeseries);
+    const dFormat = 'yyyy-MM-dd';
+    const startDate = format(new Date(data[dataStartIndex].date), dFormat);
+    const endDate = format(new Date(data[dataEndIndex].date), dFormat);
+
+    const filename = `chart.${id}.${startDate}-${endDate}`;
+    chartRef.current?.saveAsImage(filename);
```

Comment:
@nerik To achieve the date on the filename I had to do some hacky magic.
Because the dates that get exported depend on what the brush is set to I needed the brush indexes. Instead of reworking the brush state in the [Chart](https://github.com/NASA-IMPACT/delta-ui/blob/12c4c612ac444753dbcbc2ea474a370a3a3d7383/app/scripts/components/common/chart/index.tsx#L109-L117) and having to pass it all up the chain, I accessed the state of the component directly.
Not very clean, but having to pass the brush index up the chain was not super clean either. Thoughts?

Reply:
@danielfdsilva `const [brushStartIndex, setBrushStartIndex] = useState(0);` needs to move up the component tree and should live here in chart-card. Yes, it means passing down both state prop and setter down to the chart component, which is not really an issue IMHO, but if we worry about that we have a lot of tools at our disposal like context, etc.
But it is bad practice to break encapsulation and access a component internal state from outside. Let's save us some future headaches 😉
[Don't overuse refs](https://reactjs.org/docs/refs-and-the-dom.html#dont-overuse-refs)

Reply:
Instead of moving the whole state setting up the change, opted for a read only change event that doesn't break the already existing encapsulation.